pkg/cloudprovider/providers/oci/instances.go

```diff
@@ -101,7 +102,34 @@ func (cp *CloudProvider) NodeAddressesByProviderID(ctx context.Context, provider
 	if err != nil {
 		return nil, errors.Wrap(err, "GetPrimaryVNICForInstance")
 	}
-	return extractNodeAddressesFromVNIC(vnic)
+	vnicAddresses, err := extractNodeAddressesFromVNIC(vnic)
+	if err != nil {
+		return nil, err
+	}
+	addresses = append(addresses, vnicAddresses...)
+
+	if vnic != nil {
```

Comment:
Please change this to 
```
if vnic == nil {
 return addresses, nil
}
```
to reduce indentation.

Reply:
Actually, i'd imagine this check should be above extract addresses or completely omitted. Vnic should never be nil if no error was returned.